src/components/ImageGrid/ImageGridComponent.tsx

```diff
@@ -98,29 +73,147 @@ const ImageGridComponent = ({
     } else {
       setRowCol({ col: 1, row: 1 });
     }
-  }, [numberOfImgs, images]);
-
-  // to avoid bug undefined url when user change number of images
-  useEffect(() => {
-    for (let i = 0; i < numberOfImgs; i++) {
-      images[i] = exampleImages[i];
-    }
   }, [numberOfImgs]);
+  // close Carousel when user press esc
+  useEffect(() => {
+    const keydownHandler = (evnt: KeyboardEvent) => {
+      evnt.key === "Escape" && setShowCarousel(false);
+    };
+    document.addEventListener("keydown", keydownHandler);
+    return () => document.removeEventListener("keydown", keydownHandler);
+  });
 
+  const useCarousel = (index: number) => {
+    showModal && setShowCarousel(true);
+    setSelectedImgIndex(index);
+  };
+  const closeCarousel = () => {
+    setShowCarousel(false);
+  };
   return (
-    <ImageGrid
-      height={imagesGridHeight}
-      width={imagesGridWidth}
-      numberOfImgs={numberOfImgs}
-      row={rowCol.row}
-      col={rowCol.col}
+    <div
+      className="container"
+      onClick={(evnt) => {
+        // debug this using this page: https://www.reddit.com/r/typescript/comments/v5hzws/property_classname_does_not_exist_on_type/
+        const target = evnt.target as HTMLTextAreaElement;
+        const element = target.className;
+        if (element === "container") {
```

Comment:
your classname is `container`, as you define above, hence this condition is always true hence you don't need it